sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala

```diff
@@ -615,6 +615,9 @@ class MathExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(Unhex(Literal("GG")), null)
     checkEvaluation(Unhex(Literal("123")), Array[Byte](1, 35))
     checkEvaluation(Unhex(Literal("12345")), Array[Byte](1, 35, 69))
+
+    // failOnError
+    checkEvaluation(Unhex(Literal("12345"), true), Array[Byte](1, 35, 69))
```

Comment:
Does this check that compilation doesn't fail, but not the error?

Reply:
Yes this is just a positive test case that evaluation still works with failOnError enabled. It looked like there were other tests (that you commented on below), checking the error occurs, but probably doesn't care that it falls back to eval during the process.

Reply:
> but probably doesn't care that it falls back to eval during the process.

Could you add a test w/o fallbacks to catch the incorrect format, please.